### ISSUE TITLE
diffoscope: Depends on findutils with default names

### ISSUE THREAD

**Issue by rwhogg** [Homebrew/homebrew-core#8447]
# Please follow the general troubleshooting steps first:

- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [x] Confirmed this is problem with specific formulae and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new

### Bug reports:

diffoscope fails at runtime (it builds fine) with the following error if `findutils` is not built with default names:

```bash
% diffoscope docs docs.current
Traceback (most recent call last):
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/__main__.py", line 146, in main
    sys.exit(run_diffoscope(parsed_args))
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/__main__.py", line 122, in run_diffoscope
    parsed_args.file1, parsed_args.file2)
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/comparators/__init__.py", line 86, in compare_root_paths
    return compare_directories(path1, path2)
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/comparators/directory.py", line 95, in compare_directories
    return FilesystemDirectory(path1).compare(FilesystemDirectory(path2))
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/comparators/directory.py", line 130, in compare
    find_diff = Difference.from_command(FindAll, self.path, other.path)
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/difference.py", line 350, in from_command
    difference = Difference.from_feeder(feeder1, feeder2, path1, path2, *args, **kwargs)
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/difference.py", line 300, in from_feeder
    unified_diff = diff(feeder1, feeder2)
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/difference.py", line 269, in diff
    return run_diff(fd1, fd2, end_nl_q1, end_nl_q2)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/contextlib.py", line 89, in __exit__
    next(self.gen)
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/difference.py", line 210, in fd_from_feeder
    t.join()
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/difference.py", line 188, in join
    raise ex
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/difference.py", line 174, in run
    super().run(*args, **kwargs)
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/difference.py", line 195, in feed
    end_nl = feeder(f)
  File "/usr/local/Cellar/diffoscope/42_1/libexec/lib/python3.6/site-packages/diffoscope/difference.py", line 259, in feeder
    raise subprocess.CalledProcessError(returncode, command.cmdline(), output=command.stderr.getvalue())
subprocess.CalledProcessError: Command '['find', 'docs.current', '-printf', '%P\n']' returned non-zero exit status 1.
```

The problem appears to be the "-printf" which is not supported by the BSD find command. Note that installing `findutils` with the default settings is not sufficient; you get the exact same problem.

I would submit a PR for this, but I'm rather nervous about requiring that option for `findutils` since the doctor claims findutils and Python don't get along.

Config:
```
HOMEBREW_VERSION: 1.1.5-128-g824768e26
ORIGIN: https://github.com/Homebrew/brew
HEAD: 824768e26ff34a63c0ff809b500c2f3910b982ed
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 65690e38c8c6c3a9e3495fd409c7ff61988c5392
Core tap last commit: 54 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: octa-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.11.0 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/local/bin/python => /usr/local/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/local/bin/ruby => /usr/local/Cellar/ruby/2.4.0/bin/ruby
Java: 1.8.0_102
macOS: 10.11.6-x86_64
Xcode: N/A
CLT: 8.2.0.0.1.1480973914
X11: 2.7.11 => /opt/X11
```

Doctor:
```
Warning: Putting non-prefixed findutils in your path can cause python builds to fail.

Warning: You have unlinked kegs in your Cellar
Leaving kegs unlinked can lead to build-trouble and cause brews that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  elasticsearch@1.7
  php54
  php55

Warning: Some installed formula are missing dependencies.
You should `brew install` the missing dependencies:
  brew install node

Run `brew missing` for more details.
```

**Comment by ilovezfs** [comment]
you can use libexec/"gnubin" for this

**Comment by rwhogg** [comment]
It doesn't happen with plain files, but I can reproduce it with two empty directories:

```bash
% cd <some empty directory>
% mkdir a
% mkdir b
% diffoscope a b
```

**Comment by ilovezfs** [comment]
Closed via #8622.
